File: Books.Web/ViewModels/AuthorViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Books.Web.ViewModels
{
    public class AuthorViewModel
    {
	    public Guid Id { get; set; }

        [Required]
        [MaxLength(20)]
        public string FirstName { get; set; }

        [Required]
        [MaxLength(20)]
        public string LastName { get; set; }

        public Guid BookId { get; set; }
    }
}

File: Books.WebApp/Scripts/App/ViewModels/AuthorViewModel.js
"use strict";

var AuthorViewModel = function(author) {
	var self = this;

    self.id = ko.observable(author.id);

    self.firstName = ko.observable(author.firstName);
    self.firstName.extend({
        required: true,
        maxLength: 20
    });

    self.lastName = ko.observable(author.lastName);
    self.lastName.extend({
        required: true,
        maxLength: 20
    });

    self.bookId = ko.observable(author.bookId);

    self.update = function (updAuthor) {
        if (updAuthor == null) return;

        self.firstName(updAuthor.firstName);
        self.lastName(updAuthor.lastName);
    }
}